### DISCUSSION TITLE
How can we configure nextjs so that hydration does not defer TTI that much?

### DISCUSSION THREAD

**Discussion by svassaux** [vercel/next.js#20454] [Help]
I have a website with static export configured
It loads pretty fast

However, my google metrics are not very good, because of nextscripts : if I deactivate nextscripts, I reach 100 performance score on google insights.

Here, I have only 57 on mobile. There is a 8.2sec Time To Interactive.
I have already broke down scripts with dynamic imports.

Here is the profiling of my page with the performance tab and the link of my site : [melba](https://melba.io)
![image](https://user-images.githubusercontent.com/9137903/103085605-77d05700-45e2-11eb-9d4d-9541a88a4a78.png)

Can someone guide me towards what might be done wrong here ?

**Comment by awareness481** [comment]
It's not Nextscripts, you're using HTTP/1. You need to be using HTTP/2

**Reply by svassaux** [reply]
http2 is not yet not supported by heroku, which I am using
can you explain why it's a http1/2 matter ? 
If I disable nextscripts, I get a 100 perf score given the fact I am using static generation

**Comment by svassaux** [comment]
Ok, now I figure out it's linked to rehydration, I have found a document about this here : https://addyosmani.com/blog/rehydration/

So my question becomes: how can we optimize rehydration to not deffer TTI?

**Reply by JustFly1984** [reply]
I guess that the answer is on hold, till server components released.

**Reply by JustFly1984** [reply]
http2 was not supported by heroku 3 years ago, and looks like it will never be supported. http3 is on horizon, so I would recommend to switch your hosting provider, for netlify for example, they had announced next.js support last month, though you can find several hosting companies supporting next.js automated deploy.

**Comment by svassaux** [comment]
I found solutions for this problem !

You can look at this optimization todo: https://www.moovweb.com/post/lighthouse6-website-frontend-optimization

And I implemented another Lazy component than the one adviced: https://www.npmjs.com/package/react-hydration-on-demand

And here it is with 4 seconds less in script execution and a far better breakdown
![image](https://user-images.githubusercontent.com/9137903/103161516-407cc880-47e3-11eb-8f6d-59053fac9ff1.png)

**Reply by nuanyang233** [reply]
Hi, how is your site optimized for LCP？I had the same problem as you

**Reply by guitheengineer** [reply]
> Hi, how is your site optimized for LCP？I had the same problem as you

I solved a problem with my LCP (an image that took some time to finish rendering) by passing the `priority` prop. Doing this made nextjs prioritize the img and load it faster

see:
https://nextjs.org/docs/basic-features/image-optimization#priority

**Comment by benzizoo** [comment]
I personally use https://github.com/hadeeb/react-lazy-hydration
wonder if it's better, any idea?
